### PR TITLE
LWDEV-9038 [Oauth] Track user actions with session at Oauth service

### PR DESCRIPTION
**Pull request by MikhailGerasimov**

### CODE REVIEW

File: src/Core/Extensions/OpenIdConnectConstantsExt.cs [hunk 1]
```diff
@@ -7,6 +7,7 @@ public static class Claims
             public const string Country = "country";
             public const string Documents = "documents";
             public const string SignType = "SignType";
+            public const string SessionId = "http://lykke.com/oauth/sessionid";
```
Comment (simpllfy) [inline src/Core/Extensions/OpenIdConnectConstantsExt.cs]: Is it a claim name? Why is it so custom? I suggest "sid" like here
https://openid.net/specs/openid-connect-backchannel-1_0.html#Backchannel

File: src/Lykke.Service.OAuth/Controllers/AuthenticationController.cs [hunk 2]
```diff
@@ -343,11 +357,12 @@ public async Task<RegistrationResultModel> CompleteRegistration([FromBody]SignUp
                 return regResult;
             }
 
-            var identity = await _userManager.CreateUserIdentityAsync(result.Account.Id, result.Account.Email, model.Email, true);
+                await _profileActionHandler.UpdatePersonalInformation(result.Account.Id, model.FirstName, model.LastName);
+
+                var identity = await _userManager.CreateUserIdentityAsync(result.Account.Id, model.Email, model.Email, result.Token, true);
 
-            await HttpContext.SignInAsync(OpenIdConnectConstantsExt.Auth.DefaultScheme, new ClaimsPrincipal(identity));
+                await HttpContext.SignInAsync(OpenIdConnectConstantsExt.Auth.DefaultScheme, new ClaimsPrincipal(identity), new AuthenticationProperties());
```
Comment (simpllfy) [inline src/Lykke.Service.OAuth/Controllers/AuthenticationController.cs]: Why this change happened?

File: src/Lykke.Service.OAuth/Controllers/AuthenticationController.cs [hunk 3]
```diff
@@ -343,11 +357,12 @@ public async Task<RegistrationResultModel> CompleteRegistration([FromBody]SignUp
                 return regResult;
             }
 
-            var identity = await _userManager.CreateUserIdentityAsync(result.Account.Id, result.Account.Email, model.Email, true);
+                await _profileActionHandler.UpdatePersonalInformation(result.Account.Id, model.FirstName, model.LastName);
```
Comment (simpllfy) [inline src/Lykke.Service.OAuth/Controllers/AuthenticationController.cs]: Flow was: 
1. Signin
2. UpdatePersonalInformation

so it you failed at signin, update will be unreachable
Now the order is reversed, can it be a problem?

File: src/Lykke.Service.OAuth/Controllers/AuthenticationController.cs [hunk 4]
```diff
@@ -367,14 +382,17 @@ public async Task<RegistrationResultModel> CompleteRegistration([FromBody]SignUp
...
         {
-            await HttpContext.SignOutAsync(OpenIdConnectConstantsExt.Auth.DefaultScheme);
-            await HttpContext.SignOutAsync(OpenIdConnectServerDefaults.AuthenticationScheme);
-            return SignOut(OpenIdConnectConstantsExt.Auth.DefaultScheme, OpenIdConnectServerDefaults.AuthenticationScheme);
+            var sessionId = User.FindFirst(OpenIdConnectConstantsExt.Claims.SessionId)?.Value;
+            if (sessionId != null)
+            {
+                await _clientSessionsClient.DeleteSessionIfExistsAsync(sessionId);
+        }
+            return SignOut(OpenIdConnectConstantsExt.Auth.DefaultScheme);
```
Comment (simpllfy) [inline src/Lykke.Service.OAuth/Controllers/AuthenticationController.cs]: Is this call enough? Sign out call for ASOS scheme is not required anymore?

File: src/Lykke.Service.OAuth/Controllers/AuthorizationController.cs [hunk 5]
```diff
@@ -135,7 +137,7 @@ public async Task<IActionResult> Accept(CancellationToken cancellationToken)
             var scopes = request.GetScopes().ToList();
             var claims = HttpContext.User.Claims;
 
-            var identity = _userManager.CreateIdentity(scopes, claims);
+            var identity = _userManager.CreateIdentity(scopes, claims.Where(c => c.Type != OpenIdConnectConstantsExt.Claims.SessionId));
```
Comment (simpllfy) [inline src/Lykke.Service.OAuth/Controllers/AuthorizationController.cs]: Looks like it can be responsibility of UserManager

File: src/Core/Extensions/OpenIdConnectConstantsExt.cs (hunk 1 above)
Comment (MikhailGerasimov) [reply src/Core/Extensions/OpenIdConnectConstantsExt.cs]: https://openid.net/specs/openid-connect-core-1_0.html#AdditionalClaims
Collision safe means with a namespace. Actually Microsoft does the same.
https://docs.microsoft.com/en-us/dotnet/framework/wcf/extending/how-to-create-a-custom-claim

File: src/Lykke.Service.OAuth/Controllers/AuthenticationController.cs (hunk 2 above)
Comment (MikhailGerasimov) [reply src/Lykke.Service.OAuth/Controllers/AuthenticationController.cs]: Intentionally. Should be reverted

File: src/Lykke.Service.OAuth/Controllers/AuthenticationController.cs (hunk 4 above)
Comment (MikhailGerasimov) [reply src/Lykke.Service.OAuth/Controllers/AuthenticationController.cs]: There is no ASOS scheme in the authentication controller. Only Cookies must be used

File: src/Lykke.Service.OAuth/Controllers/AuthorizationController.cs (hunk 5 above)
Comment (MikhailGerasimov) [reply src/Lykke.Service.OAuth/Controllers/AuthorizationController.cs]: Here https://github.com/LykkeCity/Lykke.Service.OAuth/pull/65/files#diff-01e8016c319473d0712c0d1a7106bbb4R172 we include this claim only into access_token

File: src/Lykke.Service.OAuth/Controllers/AuthenticationController.cs (hunk 3 above)
Comment (MikhailGerasimov) [reply src/Lykke.Service.OAuth/Controllers/AuthenticationController.cs]: I don't know cases why we could not signin. We call no external services and the content of the arguments is constant.
